### ISSUE TITLE
TLS v1.3 and connection issue

### ISSUE THREAD

**Issue by treepleks** [kservices/gBridge#8]
Thanks for gBridge. I have been trying to connect my mosquitto broker to the mqtt broker of gbridge but seem to have 2 issues:

1) the gbridge profile page at https://gbridge.kappelt.net/profile says that tls v1.3 must be used. But when I set bridge_tls_version tlsv1.3 I have an error message saying that v1.3 is not supported. The error message disappears when I use tlsv1.2. Wrong profile message?

2) I can ping mqtt.gbridge.kappelt.net but the connection seems to have issues. I get messages such as:

> 1540078332: Bridge local.odroidule.kappelt-gbridge doing local SUBSCRIBE on topic gBridge/u81/+/+
> 1540078332: Bridge local.odroidule.kappelt-gbridge doing local SUBSCRIBE on topic gBridge/u81/+/+/set
> 1540078332: Connecting bridge kappelt-gbridge (mqtt.gbridge.kappelt.net:8883)
> 1540078332: Bridge odroidule.kappelt-gbridge sending CONNECT
> 1540078332: Received CONNACK on connection local.odroidule.kappelt-gbridge.
> 1540078332: Bridge local.odroidule.kappelt-gbridge sending SUBSCRIBE (Mid: 2, Topic: gBridge/u81/+/+, QoS: 0)
> 1540078332: Bridge local.odroidule.kappelt-gbridge sending SUBSCRIBE (Mid: 3, Topic: gBridge/u81/+/+/set, QoS: 0)
> 1540078333: Received PUBACK from local.odroidule.kappelt-gbridge (Mid: 1)
> 1540078333: Received SUBACK from local.odroidule.kappelt-gbridge
> 1540078333: Socket error on client local.odroidule.kappelt-gbridge, disconnecting.

Any idea ?

Regards

**Comment by treepleks** [comment]
Thanks. I should have dug the issues. Apparently, my mosquitto (1.5.3) defaults to mqttv311 as its bridge protocol. Adding 

`bridge_protocol_version mqttv31`

did it. You probably should add this line in your example to make it robust to all mosquitto variants.
Thanks again!

**Comment by peterkappelt** [comment]
Yeah, sorry, I intentionally "downgraded" the server to MQTT V3.1 - in order for it to be backward compatible. However, as you've seen, I forgot the notice in the documentation. Should be alright now.
